src/lib/y2storage/autoinst_proposal.rb

```diff
@@ -0,0 +1,184 @@
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2storage/storage_manager"
+require "y2storage/disk_analyzer"
+require "y2storage/proposal"
+require "y2storage/exceptions"
+
+module Y2Storage
+  # Class to calculate a storage proposal for autoinstallation
+  #
+  # @example Example
+  #   profile = Yast::Profile.current["partitioning"]
+  #   proposal = Y2Storage::AutoinstProposal.new(partitioning)
+  #   proposal.proposed?            # => false
+  #   proposal.proposed_devicegraph # => nil
+  #   proposal.planned_devices      # => nil
+  #
+  #   proposal.propose              # => Performs the calculation
+  #
+  #   proposal.proposed?            # => true
+  #   proposal.proposed_devicegraph # => Proposed layout
+  class AutoinstProposal
```

Comment:
This code contains AutoYaST specific code, so I do not think it will be reusable at all.